flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -54,8 +54,12 @@
 	
 
 	/** Stores the concrete key/value pairs of this configuration object. */
-	private final HashMap<String, Object> confData;
-	
+	private final Map<String, Object> confData;
+
+	/** Stores the program version */
+	public final String FLINK_VERSION = getClass().getPackage().getImplementationVersion() ==
+		null ? "FLINK_TEST_VERSION" : getClass().getPackage().getImplementationVersion();
```

Comment:
Ah yes. I had not considered this. In this case, the client version would be null.

As for ensuring that compatible versions still work, you're right. This is why I'm not in favor of using the current version string. A better approach would be to have specific release versions in the configuration file, from where everything starts.`